### ISSUE TITLE
Logstash crashing when parsing execution reports

### ISSUE THREAD

**Issue by GJB-ICM** [connamara/logstash-filter-fix_protocol#82]
Hi,

Perhaps you have experienced the following ... When processing execution reports through the filter logstash crashes and restarts. Initially I considered this may be my FIX messages but testing with your example FIX caused the same issue

Thus far I have successfully processed the following message types (All FIX 4.2): 
MsgType: "Heartbeat"
MsgType: "TestRequest"
MsgType: "Logon"
MsgType: "ResendRequest"
MsgType: "SequenceReset"
 
log-file output:
[2017-07-14T15:31:30,778][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2017-07-14T15:31:30,793][DEBUG][logstash.inputs.file     ] writing sincedb (delta since last write = 1500042690)
[2017-07-14T15:31:30,801][DEBUG][logstash.pipeline        ] filter received {"event"=>{"path"=>"/usr/share/logstash/spiderlogs/spider.test.fix.messages.log", "@timestamp"=>2017-07-14T14:31:30.789Z, "@version"=>"1", "host"=>"uk1pvrpt01", "message"=>"8=FIX.4.4\u00019=0996\u000135=8\u000149=FOOO\u000156=BAR\u000134=1505\u0001128=BAZ\u0001115=BLAH\u000152=20170620-19:22:00\u000160=20170620-19:22:00\u0001150=F\u000131=9999.11111\u0001151=0\u0001541=20171109\u000132=150000000\u0001662=1.179999948\u0001423=4\u0001663=4\u000164=20170621\u00015975=FOOBAR:20170620:1073:6\u00016=9999.11111\u000137=FOOBAR:20170620:1072:6\u000138=150000000\u0001218=14.2591147256\u0001548=FOOBAR:20170620:1073:6\u000139=2\u0001159=0\u0001669=9999.11111\u0001699=912796MB2\u0001460=6\u0001761=1\u0001223=0\u000114=925000000\u000115=USD\u000175=20170620\u0001106=TREASURY BILL \u000117=FOOBAR:20170620:1072:6:12\u0001167=TBILL\u000148=912796KX6\u0001198=7771:20170620:1072:6\u0001470=US\u0001381=123456789\u000122=1\u000154=2\u00017014=21\u000155=[N/A]\u0001118=123456789\u0001453=4\u0001448=FOOBAR\u0001447=D\u0001452=1\u0001448=JDR4:282205\u0001447=D\u0001452=11\u0001802=2\u0001523=27\u0001803=4\u0001523=25906\u0001803=4000\u0001448=ACME CORPORATION\u0001447=D\u0001452=13\u0001448=BBUNNY:13785105\u0001447=D\u0001452=36\u000110=001\u0001"}}
[2017-07-14T15:31:30,805][DEBUG][logstash.filters.grok    ] Running grok filter {:event=>2017-07-14T14:31:30.789Z uk1pvrpt01 8=FIX.4.4^A9=0996^A35=8^A49=FOOO^A56=BAR^A34=1505^A128=BAZ^A115=BLAH^A52=20170620-19:22:00^A60=20170620-19:22:00^A150=F^A31=9999.11111^A151=0^A541=20171109^A32=150000000^A662=1.179999948^A423=4^A663=4^A64=20170621^A5975=FOOBAR:20170620:1073:6^A6=9999.11111^A37=FOOBAR:20170620:1072:6^A38=150000000^A218=14.2591147256^A548=FOOBAR:20170620:1073:6^A39=2^A159=0^A669=9999.11111^A699=912796MB2^A460=6^A761=1^A223=0^A14=925000000^A15=USD^A75=20170620^A106=TREASURY BILL ^A17=FOOBAR:20170620:1072:6:12^A167=TBILL^A48=912796KX6^A198=7771:20170620:1072:6^A470=US^A381=123456789^A22=1^A54=2^A7014=21^A55=[N/A]^A118=123456789^A453=4^A448=FOOBAR^A447=D^A452=1^A448=JDR4:282205^A447=D^A452=11^A802=2^A523=27^A803=4^A523=25906^A803=4000^A448=ACME CORPORATION^A447=D^A452=13^A448=BBUNNY:13785105^A447=D^A452=36^A10=001^A}
[2017-07-14T15:31:30,814][DEBUG][logstash.filters.grok    ] Event now:  {:event=>2017-07-14T14:31:30.789Z uk1pvrpt01 8=FIX.4.4^A9=0996^A35=8^A49=FOOO^A56=BAR^A34=1505^A128=BAZ^A115=BLAH^A52=20170620-19:22:00^A60=20170620-19:22:00^A150=F^A31=9999.11111^A151=0^A541=20171109^A32=150000000^A662=1.179999948^A423=4^A663=4^A64=20170621^A5975=FOOBAR:20170620:1073:6^A6=9999.11111^A37=FOOBAR:20170620:1072:6^A38=150000000^A218=14.2591147256^A548=FOOBAR:20170620:1073:6^A39=2^A159=0^A669=9999.11111^A699=912796MB2^A460=6^A761=1^A223=0^A14=925000000^A15=USD^A75=20170620^A106=TREASURY BILL ^A17=FOOBAR:20170620:1072:6:12^A167=TBILL^A48=912796KX6^A198=7771:20170620:1072:6^A470=US^A381=123456789^A22=1^A54=2^A7014=21^A55=[N/A]^A118=123456789^A453=4^A448=FOOBAR^A447=D^A452=1^A448=JDR4:282205^A447=D^A452=11^A802=2^A523=27^A803=4^A523=25906^A803=4000^A448=ACME CORPORATION^A447=D^A452=13^A448=BBUNNY:13785105^A447=D^A452=36^A10=001^A}
[2017-07-14T15:31:30,859][ERROR][logstash.pipeline        ] Exception in pipelineworker, the pipeline stopped processing new events, please check your filter configuration and restart Logstash. {"exception"=>"undefined method `to_hash' for #<Java::Quickfix::Group:0x2327e6e9>", "backtrace"=>["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:87:in `field_map_to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:72:in `field_map_to_hash'", "org/jruby/RubyRange.java:476:in `each'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/activesupport-4.1.16/lib/active_support/core_ext/range/each.rb:7:in `each_with_time_with_zone'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:69:in `field_map_to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:25:in `to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_protocol.rb:52:in `filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145:in `do_filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164:in `multi_filter'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161:in `multi_filter'", "/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43:in `multi_filter'", "(eval):382:in `initialize'", "org/jruby/RubyArray.java:1613:in `each'", "(eval):379:in `initialize'", "org/jruby/RubyProc.java:281:in `call'", "(eval):184:in `filter_func'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:383:in `filter_batch'", "org/jruby/RubyProc.java:281:in `call'", "/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:238:in `each'", "org/jruby/RubyHash.java:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:237:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:382:in `filter_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:363:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:330:in `start_workers'"]}
[2017-07-14T15:31:31,045][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NoMethodError: undefined method `to_hash' for #<Java::Quickfix::Group:0x2327e6e9>>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:87:in `field_map_to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:72:in `field_map_to_hash'", "org/jruby/RubyRange.java:476:in `each'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/activesupport-4.1.16/lib/active_support/core_ext/range/each.rb:7:in `each_with_time_with_zone'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:69:in `field_map_to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_message.rb:25:in `to_hash'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/lib/logstash/filters/fix_protocol.rb:52:in `filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:145:in `do_filter'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:164:in `multi_filter'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/filters/base.rb:161:in `multi_filter'", "/usr/share/logstash/logstash-core/lib/logstash/filter_delegator.rb:43:in `multi_filter'", "(eval):382:in `initialize'", "org/jruby/RubyArray.java:1613:in `each'", "(eval):379:in `initialize'", "org/jruby/RubyProc.java:281:in `call'", "(eval):184:in `filter_func'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:383:in `filter_batch'", "org/jruby/RubyProc.java:281:in `call'", "/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:238:in `each'", "org/jruby/RubyHash.java:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:237:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:382:in `filter_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:363:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:330:in `start_workers'"]}
[2017-07-14T15:31:43,360][DEBUG][logstash.plugins.registry] Adding plugin to the registry {:name=>"fb_apache", :type=>:modules, :class=>#<LogStash::Modules::Scaffold:0x5296f15c @module_name="fb_apache", @directory="/usr/share/logstash/modules/fb_apache/configuration">}
[2017-07-14T15:31:43,412][DEBUG][logstash.runner          ] -------- Logstash Settings (* means modified) ---------

Conf file:
input {
  file {
    path => "/usr/share/logstash/spiderlogs/spider.*"
    start_position => "beginning"
  }
}
filter {
  grok {
    match => ["message","%{GREEDYDATA:fix_string}"]
  }
  if [fix_string] =~ "=FIX.4.2" {
    fix_protocol {
      fix_message => fix_string
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX42.xml"
    }
  }
  if [fix_string] =~ "=FIX.4.3" {
    fix_protocol {
      fix_message => fix_string
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX43.xml"
    }
  }
  if [fix_string] =~ "=FIX.4.4" {
    fix_protocol {
      fix_message => fix_string
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX44.xml"
    }
  }
  if [fix_string] =~ "=FIX.5.0" {
    fix_protocol {
      fix_message => fix_string
      session_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIXT11.xml"
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX50SP1.xml"
    }
  }
  else {
    drop {}
  }
}
output {
  elasticsearch {
    hosts => "127.0.0.1:9200"
    index => "logstash-betafix-index"
  }
  stdout { codec => rubydebug  }
}

Please excuse my lack of ruby/logstash knowledge ... Any assistance would be gratefully received.

Thanks

**Comment by daino3** [comment]
Hi @GJB-ICM - it looks like you're using version `0.3.1`. I [recently fixed](https://github.com/connamara/logstash-filter-fix_protocol/pull/78) a FIX 44 parsing bug. 

Please update to `0.3.2` with the following command:
+ `logstash-plugin update logstash-filter-fix_protocol`

and ensure `0.3.2` is installed with:
+ `logstash-plugin list --verbose`

**Comment by GJB-ICM** [comment]
Hi @daino3 

Thank you for the response ... I updated to v0.3.2 (my Logstash version is 5.5.0)

This successfully parsed the example FIX 4.4 message but crashed with one of mine:
8=FIX.4.4^A9=415^A35=AE^A34=217^A49=FX-TCR^A56=MY-TCR^A52=20160327-21:22:52.385^A571=EWTVOOZV44394301999999981^A568=AD_20160327-190002^A150=F^A39=2^A17=Z000RO57DS^A570=N^A55=EUR/USD^A167=FOR^A38=82000.000000^A6054=91594.000000^A44=1.117000^A32=82000.000000^A31=1.117000^A194=1.117000^A195=0.0000^A75=20160328^A6215=SP^A60=20160327-19:12:52^A552=1^A54=2^A37=Z000RO57DS^A453=1^A448=myuserid^A447=D^A452=11^A1=MY-ACCOUNT G/U^A15=EUR^A40=D^A487=0^A64=20160330^A10=095^A

Log contents:
[2017-07-17T07:45:27,781][INFO ][logstash.agent           ] Successfully started Logstash API endpoint {:port=>9600}
[2017-07-17T07:45:27,812][DEBUG][logstash.inputs.file     ] writing sincedb (delta since last write = 1500273927)
[2017-07-17T07:45:27,823][DEBUG][logstash.pipeline        ] filter received {"event"=>{"path"=>"/usr/share/logstash/spiderlogs/spider.test.fix.messages.log", "@timestamp"=>2017-07-17T06:45:27.801Z, "@version"=>"1", "host"=>"uk1pvrpt01", "message"=>"8=FIX.4.4\u00019=415\u000135=AE\u000134=217\u000149=FX-TCR\u000156=MY-TCR\u000152=20160327-21:22:52.385\u0001571=EWTVOOZV44394301999999981\u0001568=AD_20160327-190002\u0001150=F\u000139=2\u000117=Z000RO57DS\u0001570=N\u000155=EUR/USD\u0001167=FOR\u000138=82000.000000\u00016054=91594.000000\u000144=1.117000\u000132=82000.000000\u000131=1.117000\u0001194=1.117000\u0001195=0.0000\u000175=20160328\u00016215=SP\u000160=20160327-19:12:52\u0001552=1\u000154=2\u000137=Z000RO57DS\u0001453=1\u0001448=myuserid\u0001447=D\u0001452=11\u00011=MY-ACCOUNT G/U\u000115=EUR\u000140=D\u0001487=0\u000164=20160330\u000110=095\u0001"}}
[2017-07-17T07:45:27,824][DEBUG][logstash.filters.grok    ] Running grok filter {:event=>2017-07-17T06:45:27.801Z uk1pvrpt01 8=FIX.4.4^A9=415^A35=AE^A34=217^A49=FX-TCR^A56=MY-TCR^A52=20160327-21:22:52.385^A571=EWTVOOZV44394301999999981^A568=AD_20160327-190002^A150=F^A39=2^A17=Z000RO57DS^A570=N^A55=EUR/USD^A167=FOR^A38=82000.000000^A6054=91594.000000^A44=1.117000^A32=82000.000000^A31=1.117000^A194=1.117000^A195=0.0000^A75=20160328^A6215=SP^A60=20160327-19:12:52^A552=1^A54=2^A37=Z000RO57DS^A453=1^A448=myuserid^A447=D^A452=11^A1=MY-ACCOUNT G/U^A15=EUR^A40=D^A487=0^A64=20160330^A10=095^A}
[2017-07-17T07:45:27,829][DEBUG][logstash.filters.grok    ] Event now:  {:event=>2017-07-17T06:45:27.801Z uk1pvrpt01 8=FIX.4.4^A9=415^A35=AE^A34=217^A49=FX-TCR^A56=MY-TCR^A52=20160327-21:22:52.385^A571=EWTVOOZV44394301999999981^A568=AD_20160327-190002^A150=F^A39=2^A17=Z000RO57DS^A570=N^A55=EUR/USD^A167=FOR^A38=82000.000000^A6054=91594.000000^A44=1.117000^A32=82000.000000^A31=1.117000^A194=1.117000^A195=0.0000^A75=20160328^A6215=SP^A60=20160327-19:12:52^A552=1^A54=2^A37=Z000RO57DS^A453=1^A448=myuserid^A447=D^A452=11^A1=MY-ACCOUNT G/U^A15=EUR^A40=D^A487=0^A64=20160330^A10=095^A}
[2017-07-17T07:45:27,868][DEBUG][logstash.pipeline        ] output received {"event"=>{"6054"=>"91594.000000", "BeginString"=>"FIX.4.4", "SenderCompID"=>"FX-TCR", "SecurityType"=>"FOREIGN_EXCHANGE_CONTRACT", "unknown_fields"=>["6054", "6215"], "ExecID"=>"Z000RO57DS", "OrderQty"=>82000.0, "Symbol"=>"EUR/USD", "TargetCompID"=>"MY-TCR", "NoSides"=>[{"Side"=>"SELL", "Account"=>"MY-ACCOUNT G/U", "OrdType"=>"PREVIOUSLY_QUOTED", "Currency"=>"EUR", "NoPartyIDs"=>[{"PartyRole"=>11, "PartyID"=>"myuserid", "PartyIDSource"=>"PROPRIETARY_CUSTOM_CODE"}], "OrderID"=>"Z000RO57DS"}], "LastQty"=>82000.0, "TradeRequestID"=>"AD_20160327-190002", "LastSpotRate"=>1.117, "TransactTime"=>"20160327-19:12:52", "PreviouslyReported"=>false, "path"=>"/usr/share/logstash/spiderlogs/spider.test.fix.messages.log", "SendingTime"=>"20160327-21:22:52.385", "OrdStatus"=>"FILLED", "SettlDate"=>"20160330", "@version"=>"1", "host"=>"uk1pvrpt01", "TradeDate"=>"20160328", "ExecType"=>"TRADE", "TradeReportTransType"=>0, "fix_string"=>"8=FIX.4.4\u00019=415\u000135=AE\u000134=217\u000149=FX-TCR\u000156=MY-TCR\u000152=20160327-21:22:52.385\u0001571=EWTVOOZV44394301999999981\u0001568=AD_20160327-190002\u0001150=F\u000139=2\u000117=Z000RO57DS\u0001570=N\u000155=EUR/USD\u0001167=FOR\u000138=82000.000000\u00016054=91594.000000\u000144=1.117000\u000132=82000.000000\u000131=1.117000\u0001194=1.117000\u0001195=0.0000\u000175=20160328\u00016215=SP\u000160=20160327-19:12:52\u0001552=1\u000154=2\u000137=Z000RO57DS\u0001453=1\u0001448=myuserid\u0001447=D\u0001452=11\u00011=MY-ACCOUNT G/U\u000115=EUR\u000140=D\u0001487=0\u000164=20160330\u000110=095\u0001", "BodyLength"=>"415", "LastPx"=>1.117, "TradeReportID"=>"EWTVOOZV44394301999999981", "MsgType"=>"TradeCaptureReport", "message"=>"8=FIX.4.4\u00019=415\u000135=AE\u000134=217\u000149=FX-TCR\u000156=MY-TCR\u000152=20160327-21:22:52.385\u0001571=EWTVOOZV44394301999999981\u0001568=AD_20160327-190002\u0001150=F\u000139=2\u000117=Z000RO57DS\u0001570=N\u000155=EUR/USD\u0001167=FOR\u000138=82000.000000\u00016054=91594.000000\u000144=1.117000\u000132=82000.000000\u000131=1.117000\u0001194=1.117000\u0001195=0.0000\u000175=20160328\u00016215=SP\u000160=20160327-19:12:52\u0001552=1\u000154=2\u000137=Z000RO57DS\u0001453=1\u0001448=myuserid\u0001447=D\u0001452=11\u00011=MY-ACCOUNT G/U\u000115=EUR\u000140=D\u0001487=0\u000164=20160330\u000110=095\u0001", "MsgSeqNum"=>"217", "tags"=>["_fix_field_not_found"], "@timestamp"=>2017-07-17T06:45:27.801Z, "Price"=>1.117, "CheckSum"=>"095", "6215"=>"SP", "LastForwardPoints"=>"0.0000"}}
[2017-07-17T07:45:28,125][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NameError: undefined local variable or method `dotfile' for #<AwesomePrint::Inspector:0x356e2a36>>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/awesome_print-1.8.0/lib/awesome_print/inspector.rb:163:in `merge_custom_defaults!'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/awesome_print-1.8.0/lib/awesome_print/inspector.rb:50:in `initialize'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/awesome_print-1.8.0/lib/awesome_print/core_ext/kernel.rb:9:in `ai'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-rubydebug-3.0.3/lib/logstash/codecs/rubydebug.rb:39:in `encode_default'", "org/jruby/RubyMethod.java:120:in `call'", "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-codec-rubydebug-3.0.3/lib/logstash/codecs/rubydebug.rb:35:in `encode'", "/usr/share/logstash/logstash-core/lib/logstash/codecs/base.rb:50:in `multi_encode'", "org/jruby/RubyArray.java:1613:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/codecs/base.rb:50:in `multi_encode'", "/usr/share/logstash/logstash-core/lib/logstash/outputs/base.rb:90:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/single.rb:15:in `multi_receive'", "org/jruby/ext/thread/Mutex.java:149:in `synchronize'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator_strategies/single.rb:14:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/output_delegator.rb:47:in `multi_receive'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:420:in `output_batch'", "org/jruby/RubyHash.java:1342:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:419:in `output_batch'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:365:in `worker_loop'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:330:in `start_workers'"]}
[2017-07-17T07:45:39,256][DEBUG][logstash.plugins.registry] Adding plugin to the registry {:name=>"fb_apache", :type=>:modules, :class=>#<LogStash::Modules::Scaffold:0x46131545 @module_name="fb_apache", @directory="/usr/share/logstash/modules/fb_apache/configuration">}
[2017-07-17T07:45:39,284][DEBUG][logstash.runner          ]
 -------- Logstash Settings (* means modified) --------- (Logstash reloading)



It may be worth noting I get crashes with FIX 4.2 as well for any message type other than admin/Heartbeat messages.

Conf file (Updated):
input {
  file {
    path => "/usr/share/logstash/spiderlogs/spider.*"
    start_position => "beginning"
  }
}
filter {
  grok {
    match => ["message","%{GREEDYDATA:fix_string}"]
  }
  if [message] =~ "=FIX.4.2" {
    fix_protocol {
      fix_message => message
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX42.xml"
    }
  }
  if [message] =~ "=FIX.4.3" {
    fix_protocol {
      fix_message => message
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX43.xml"
    }
  }
  if [message] =~ "=FIX.4.4" {
    fix_protocol {
      fix_message => message
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX44.xml"
    }
  }
  if [message] =~ "=FIX.5.0" {
    fix_protocol {
      fix_message => message
      session_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIXT11.xml"
      data_dictionary_path => "/usr/share/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-fix_protocol-0.3.1/spec/fixtures/FIX50SP1.xml"
    }
  }
}
output {
  elasticsearch {
    hosts => "127.0.0.1:9200"
    index => "logstash-betafix-index"
  }
  stdout { codec => rubydebug  }
}

Thanks

**Comment by daino3** [comment]
Hi @GJB-ICM - the error being thrown for you is happening because of another gem `awesome_print` seen here in your stack trace:

```
[2017-07-17T07:45:28,125][FATAL][logstash.runner ] An unexpected error occurred! {:error=>#<NameError: undefined local variable or method dotfile' for #<AwesomePrint::Inspector:0x356e2a36>>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/1.9/gems/awesome_print-1.8.0/lib/awesome_print/inspector.rb:163:inmerge_custom_defaults!'", 
```

This appears to be an error in their source code [here](https://github.com/awesome-print/awesome_print/blob/f60f9e82e26d3a22d4b7c58d7090b75d6687d90c/lib/awesome_print/inspector.rb#L164)

Can you investigate that first please? If you can get an actual error being thrown by the FIX filter, I will investigate that error. 

Thanks.

**Comment by GJB-ICM** [comment]
Hi @daino3 ,

Apologies assumed that it was an error driven by data passed from this filter. (https://github.com/awesome-print/awesome_print/issues/316)

I ran bin/logstash-plugin update

It updated the following plugins:
Updated logstash-codec-collectd 3.0.5 to 3.0.6
Updated logstash-codec-netflow 3.4.1 to 3.5.0
Updated logstash-filter-dissect 1.0.9 to 1.0.11
Updated logstash-filter-useragent 3.1.1 to 3.2.0
Updated logstash-input-beats 3.1.21 to 3.1.23
Updated logstash-input-dead_letter_queue 1.0.4 to 1.0.5
Updated logstash-input-gelf 3.0.4 to 3.0.5
Updated logstash-input-tcp 4.1.2 to 4.2.2
Updated logstash-input-xmpp 3.1.3 to 3.1.4
Updated logstash-output-http 4.3.1 to 4.3.2
Updated logstash-output-rabbitmq 4.0.8 to 4.0.9
Updated logstash-output-xmpp 3.0.2 to 3.0.5

Since the issue is resolved.

Thank you for all the help and the awesome plugin.
